sto/ethereum/nonce.py

```diff
@@ -36,8 +39,9 @@ def restart_nonce(logger: Logger,
 
     account = service.get_or_create_broadcast_account()
     txs = service.get_last_transactions(limit=1)
-    if txs.count() > 0:
-        raise HistoryDeleteNeeded("Cannot reset nonce as the database contains txs for {}. Delete database to restart.".format(service.address))
+    if txs.count() > 0 and reset_pending_tx = True:
+        service.delete_pending_broadcasts()
```

Comment:
We can't just blindly delete pending transactions. Transactions are created and broadcasted in a particular order. 
It is quite possible that the earlier txs might succeed and the later ones will fail. In such case deleting txs and running the same command will force the creation of old ones that succeeded already. This is redundant and we should avoid it. 

We should rather find a way to update nonce of all pending transactions in the order they were created.